crates/matrix-sdk-base/src/sliding_sync.rs

```diff
@@ -371,7 +371,17 @@ impl BaseClient {
 
 fn choose_event_to_cache(events: &[SyncTimelineEvent]) -> Option<SyncTimelineEvent> {
     // Later, we will be cleverer. For now, the last event
-    events.last().cloned()
+    events
+        .iter()
+        .rfind(|&e| {
+            let timeline_event: serde_json::Result<AnySyncTimelineEvent> = e.event.deserialize();
+            if let Ok(timeline_event) = timeline_event {
+                timeline_event.event_type().to_string() == "m.room.message"
```

Comment:
I think `event_type()` should also have an `as_ref()` that derefs into a `&str`.

Reply:
Nope, but this should compare against the right enum variant instead.

Reply:
Yep, I just found that, thanks.

Reply:
Duh, what was I thinking, just use the enum :sweat_smile:.

Reply:
Fixed in 214c0387b8

Reply:
:+1: